daemon/daemon.go

```diff
@@ -159,15 +160,18 @@ func (daemon *Daemon) NewResolveOptionsFunc() resolver.ResolveOptionsFunc {
 		)
 		// must trim "https://" or "http://" prefix
 		for i, v := range daemon.configStore.Mirrors {
-			v = strings.TrimPrefix(v, "https://")
-			v = strings.TrimPrefix(v, "http://")
+			if uri, err := url.Parse(v); err == nil {
+				v = uri.Host
```

Comment:
@dmcgowan aren't mirrors allowed to have a path component ?

Reply:
Nevermind, apparently not: https://github.com/moby/moby/blob/98fc091/registry/config.go#L319

Reply:
Actually, I think this may have been over-restrictive and we supported this in the past https://github.com/moby/moby/issues/36598 😓 not sure if we should loosen that validation again?

Reply:
validation was added in https://github.com/moby/moby/pull/29650

Reply:
This PR actually fixes an issue with regular mirrors, while the issue about paths in the mirror url has been there prior. I'm fine taking in this quick fix while someone who wants to contribute fixes the issue you mention. Also, most of this code will go away with containerd integration.

Reply:
containerd does not currently support subpaths either so its not only validation

Reply:
yeah makes sense